File: Codewars/CPP/Last_even_indexed_numbers.cpp
#include <vector>

std::vector<int> evenNumbers(std::vector<int> arr, int n) {
    std::vector <int> result(n);

    for (int i = arr.size() - 1, j = n - 1; j >= 0 ; i--)
    {
        if (arr[i] % 2 == 0)
        {
            result[j] = arr[i];
            j--;
        }
    }

    return result;
}

File: Codewars/Python/Last_even_indexed_numbers.py
def even_numbers(arr,n):
    result = list("0" * n)
    i = len(arr) - 1
    j = n - 1
    while (j >= 0):
        if arr[i] % 2 == 0:
            result[j] = arr[i]
            j -= 1
        i -= 1
    return result